### ISSUE TITLE
lib/libunwind-ptrace.so has unresolved symbols: lzma_index_size and other lzma_* symbols

### ISSUE THREAD

**Issue by yurivict** [libunwind/libunwind#250]
The SimGrid project https://framagit.org/simgrid/simgrid fails to link on FreeBSD:

```
> ld: error: /usr/local/lib/libunwind-ptrace.so: undefined reference to lzma_index_size
> ld: error: /usr/local/lib/libunwind-ptrace.so: undefined reference to lzma_stream_buffer_decode
> ld: error: /usr/local/lib/libunwind-ptrace.so: undefined reference to lzma_index_end
> ld: error: /usr/local/lib/libunwind-ptrace.so: undefined reference to lzma_stream_footer_decode
> ld: error: /usr/local/lib/libunwind-ptrace.so: undefined reference to lzma_index_uncompressed_size
> ld: error: /usr/local/lib/libunwind-ptrace.so: undefined reference to lzma_index_buffer_decode
```

The downstream bug report https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=237919 was closed but the problem is still present now.

libunwind-20201110
OS: FreeBSD 12.2
clang-10

**Comment by am11** [comment]
Does it help if a line is added at https://framagit.org/simgrid/simgrid/-/blob/efb44ffc0e3b69f9e0105baf38d91deae45856c3/tools/cmake/Modules/FindLibunwind.cmake#L95
```diff
if(HAVE_LIBUNWIND)
+ # message(FATAL_ERROR "following should include lzma: ${LIBUNWIND_LIBRARIES}") # if it already does, try reordering
+  set(LIBUNWIND_LIBRARIES "-llzma ${LIBUNWIND_LIBRARIES}")
  message(STATUS "Dependencies induced by libunwind: ${LIBUNWIND_LIBRARIES}")
else()
```

this order is known to work on Linux: `-lunwind-x86_64 -llzma -lunwind`.

**Comment by bregma** [comment]
This should be fixed by the fix for #586
